### PR TITLE
Add private tangle tests CI

### PR DESCRIPTION
**Pull request by thibault-martinez**
## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: .github/workflows/coverage.yml [hunk 1]
```diff
@@ -63,10 +63,17 @@ jobs:
     - name: Install binutils and rustfilt
       run: cargo install cargo-binutils rustfilt
 
+    - name: Start private tangle
+      uses: './.github/actions/private-tangle/setup'
+
     - name: Run tests and collect coverage
       run: bash ./coverage.sh
```
Comment (Thoralf-M) [inline .github/workflows/coverage.yml:70]: Also need to add `-- --include-ignored` in coverage.sh if you want to test the ignored tests

Comment (thibault-martinez) [reply .github/workflows/coverage.yml:70]: Thanks

**Review by Thoralf-M — DISMISSED** [review]

**Review by Thoralf-M — APPROVED** [review]

File: coverage.sh [hunk 2]
```diff
@@ -27,7 +27,7 @@ BINARIES=""
 for file in \
   $( \
     RUSTFLAGS="-C instrument-coverage" \
-      cargo +nightly test --tests --all --all-features --no-run --message-format=json \
+      cargo +nightly test --tests --all --all-features --no-run --message-format=json -- --include-ignored \
         | jq -r "select(.profile.test == true) | .filenames[]" \
         | grep -v dSYM - \
   ); \
```
Comment (kwek20) [inline coverage.sh:33]: did you run this entirely? last 2 cleanup line gave me errors, thats why theyre different in wallet.rs
```
find . -name "*.profraw" -type f -delete &> /dev/null || true
find . -name "*.profdata" -type f -delete &> /dev/null || true
````

File: client/tests/node_api.rs [hunk 3]
```diff
@@ -16,9 +16,9 @@ use iota_types::block::{
 };
 
 const DEFAULT_DEVNET_NODE_URL: &str = "http://localhost:14265";
-const DEFAULT_DEVNET_FAUCET_URL: &str = "http://localhost:14265";
```
Comment (kwek20) [inline client/tests/node_api.rs:19]: should we move all these to common/mod.rs?

Comment (thibault-martinez) [reply client/tests/node_api.rs:19]: We probably need to do the same as in the wallet, but not in this PR

File: coverage.sh (hunk 2 above)
Comment (thibault-martinez) [reply coverage.sh:33]: It's been like that for a long time and working in the CI 🤷🏻

Comment (kwek20) [reply coverage.sh:33]: probably due to the sub folder we have in iota.rs, coolio

**Review by kwek20 — APPROVED** [review]

**Pull request merged**